### PR TITLE
Test Catalyst wheels with the latest LQ on macOS

### PR DESCRIPTION
**Pull request by maliasadi**

### CODE REVIEW

**Comment by maliasadi** [comment]
@dime10 @erick-xanadu we can merge this PR to #558 when it looks good.

File: runtime/Makefile [hunk 1]
```diff
@@ -15,7 +15,7 @@ ENABLE_LIGHTNING_KOKKOS?=ON
 ENABLE_OPENQASM?=ON
 ENABLE_ASAN?=OFF
 # TODO: Update to v0.35.0 after the next lightning release.
-LIGHTNING_GIT_TAG_VALUE?="2716864521c539429bd382b92151a918d1a076ce"
+LIGHTNING_GIT_TAG_VALUE?="master"
```
Comment (erick-xanadu) [inline runtime/Makefile:18]: Going what with was said in the meeting? Let's not target master and target a specific commit. Just change to the specific commit in master for the time being.

File: .github/workflows/build-wheel-macos-x86_64.yaml [hunk 2]
```diff
@@ -359,6 +359,10 @@ jobs:
       run: |
         python${{ matrix.python_version }} -m pip install $GITHUB_WORKSPACE/dist/*.whl
 
+    - name: Install PennyLane-Lightning master
+      run: |
+        python${{ matrix.python_version }} -m pip install -i https://test.pypi.org/simple/ PennyLane-Lightning --pre --upgrade
+
```
Comment (erick-xanadu) [inline .github/workflows/build-wheel-macos-x86_64.yaml:365]: Let's add these to linux as well. And let's also add the specific commit here as well.

**Comment by dime10** [comment]
This just manually installs lightning in a bunch of scripts, isn't Erick's solution much more efficient to handle it via setuptools, just like we do with pl? (#559 )

**Comment by maliasadi** [comment]
didn't see PR https://github.com/PennyLaneAI/catalyst/pull/559 :partying_face:  :partying_face: please just ensure that the same git commit is used to build LQ and LK backend devices in the runtime :100:

**Pull request closed without merging**